lib/ansible/template/__init__.py

```diff
@@ -482,6 +499,16 @@ def template(self, variable, convert_bare=False, preserve_trailing_newlines=True
             else:
                 return variable
 
+    def is_template(self, data):
+        ''' lets us know if data has a template'''
+        try:
+            new = self.template(data)
+        except UndefinedError:
+            return True
```

Comment:
I have a test case or two that fails somewhere in here, if the template string includes a '{%something%}'.
Not sure what the expected behavior is for that.

```
__________________________________________________________________________ TestTemplarIsTemplate.test_is_template_block ___________________________________________________________________________
self = <units.template.test_templar.TestTemplarIsTemplate testMethod=test_is_template_block>

    def test_is_template_block(self):
        res = self.templar.is_template(u'1 2 {%what%} 3 4 {{ some_var }} 5 6 7')
>       self.assertTrue(res)
E       AssertionError: False is not true

test/units/template/test_templar.py:276: AssertionError
_____________________________________________________________________ TestTemplarIsTemplate.test_is_template_block_undefined ______________________________________________________________________
self = <units.template.test_templar.TestTemplarIsTemplate testMethod=test_is_template_block_undefined>

    def test_is_template_block_undefined(self):
        res = self.templar.is_template(u'1 2 {%what%} 3 4 {{ foo }} 5 6 7')
>       self.assertTrue(res)
E       AssertionError: False is not true

test/units/template/test_templar.py:272: AssertionError
```